packages/engine/src/output/buffer/util.rs

```diff
@@ -1,42 +1,83 @@
-    //     } else {
-    //         Err(Error::from(format!(
-    //             "Error while trying to remove experiment parts folder: {err_string}"
-    //         )))
-    //     }
-    // })?;
+    // Cleanup python socket files in case the engine didn't
+    let frompy_files = glob::glob(&format!("{experiment_id}-frompy*"))
+        .map_err(|err| Error::Unique(format!("cleanup glob error: {err}")))?;
+    let topy_files = glob::glob(&format!("{experiment_id}-topy*"))
+        .map_err(|err| Error::Unique(format!("cleanup glob error: {err}")))?;
+
+    frompy_files
+        .chain(topy_files)
+        .filter_map(std::result::Result::ok)
+        .for_each(|path| match std::fs::remove_file(&path) {
+            Ok(_) => {
+                match exit_status {
+                    EngineExitStatus::Success => tracing::event!(
+                        tracing::Level::ERROR,
+                        "Removed file {path:?} that should've been cleanup by the engine."
+                    ),
+                    EngineExitStatus::Error => tracing::event!(
+                        tracing::Level::TRACE,
+                        "Removed file {path:?} that should've been cleanup by the engine."
+                    ),
+                };
+            }
+            Err(err) => {
+                tracing::warn!("Could not clean up {path:?}: {err}");
+            }
+        });
+
+    remove_experiment_parts(experiment_id)?;
 
     Ok(())
 }
 
-#[allow(dead_code)]
 fn remove_experiment_parts(experiment_id: &ExperimentId) -> Result<()> {
-    let mut base_path = PathBuf::from(RELATIVE_PARTS_FOLDER);
-    // TODO: this is unused at the moment so it's fine, but this logic is wrong, we name our folders
-    //  differently, we should update the design to store the paths and use them here when we use
-    //  the clean up code again
-    base_path.push(experiment_id.to_string());
-    tracing::trace!("Removing all parts files in: {base_path:?}");
-    std::fs::remove_dir_all(base_path)?;
+    let parts_fimes = glob::glob(&format!("{RELATIVE_PARTS_FOLDER}/{experiment_id}"))
+        .map_err(|err| Error::Unique(format!("cleanup glob error: {err}")))?;
+
+    parts_fimes
+        .filter_map(std::result::Result::ok)
+        .for_each(|path| match std::fs::remove_dir_all(&path) {
+            Ok(_) => {
+                tracing::trace!("Removed parts folder {path:?}.")
+            }
+            Err(err) => {
+                tracing::warn!("Could not clean up {path:?}: {err}");
+            }
+        });
+
+    // TODO: use ErrorKind::DirectoryNotEmpty when it's stable and remove the `if`
+    if std::fs::read_dir(RELATIVE_PARTS_FOLDER)?.next().is_none() {
+        match std::fs::remove_dir(RELATIVE_PARTS_FOLDER) {
+            Ok(_) => tracing::trace!("Removed parts folder."),
+            // Err(err) if err.kind() == std::io::ErrorKind::DirectoryNotEmpty => {}
+            Err(err) => {
+                return Err(err.into());
+            }
```

Comment:
I don't think we should return an error here but print a warning instead. This also would allow removing the if-statement above.

Reply:
Applied in https://github.com/hashintel/hash/pull/477/commits/7e2bcf26c3eef091683e8e5ba14c8e622acb5e33